src/qt/rpcconsole.cpp

```diff
@@ -1240,6 +1240,14 @@ void RPCConsole::updateNodeDetail(const CNodeCombinedStats *stats)
     ui->peerDirection->setText(stats->nodeStats.fInbound ? tr("Inbound") : tr("Outbound"));
     ui->peerHeight->setText(QString("%1").arg(QString::number(stats->nodeStats.nStartingHeight)));
     ui->peerWhitelisted->setText(stats->nodeStats.fWhitelisted ? tr("Yes") : tr("No"));
+    ui->peerNodeType->setText(stats->nodeStats.fMasternode ? tr("Masternode") : tr("Normal"));
+    if (stats->nodeStats.fMasternode) {
+        auto dmn = clientModel->getMasternodeList().GetMNByService(stats->nodeStats.addr);
+        ui->peerPoSeScore->setText(QString::number(dmn->pdmnState->nPoSePenalty));
+    } else {
+        ui->peerPoSeScore->setText(tr("N/A"));
+    }
+
```

Comment:
`fMasternode` is a flag to indicate if this is a special kind of _connection_ or a regular one. A proper way to check if this _node_ is a (verified) masternode or not is to use `verifiedProRegTxHash`.
```suggestion
    if (stats->nodeStats.verifiedProRegTxHash.IsNull()) {
        ui->peerNodeType->setText(tr("Normal"));
        ui->peerPoSeScore->setText(tr("N/A"));
    } else {
        ui->peerNodeType->setText(tr("Masternode"));
        auto dmn = clientModel->getMasternodeList().GetMNByService(stats->nodeStats.addr);
        ui->peerPoSeScore->setText(dmn == nullptr ? tr("N/A") : QString::number(dmn->pdmnState->nPoSePenalty));
    }
```

Reply:
Thank you for the insight :).